src/main/java/com/forms/formswebapp/form/FormService.java

```diff
@@ -31,4 +37,11 @@ void updateExpiredForms() {
         expiredForms.forEach(Form::expire);
         formRepository.saveAll(expiredForms);
     }
+
+    FormResponseDto getFormByLink(String link) {
+        Form form = formRepository.findByLink(link)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
```

Comment:
Jak chcesz to zamiast tego exceptiona możesz dodać exception handler taki jak jest dla usera `AuthenticationExceptionHandler` i zwracać ErrorResponse.

Ale to już opcjonalnie